go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -114,50 +118,66 @@ func (p daskResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsC
 	} else if taskTemplate == nil {
 		return nil, errors.Errorf(errors.BadTaskSpecification, "nil task specification")
 	}
-	defaults, err := getDefaults(ctx, taskCtx, *taskTemplate)
-	if err != nil {
-		return nil, err
-	}
-	clusterName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()
 
 	daskJob := plugins.DaskJob{}
 	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &daskJob)
 	if err != nil {
 		return nil, errors.Wrapf(errors.BadTaskSpecification, err, "invalid TaskSpecification [%v], failed to unmarshal", taskTemplate.GetCustom())
 	}
 
-	workerSpec, err := createWorkerSpec(*daskJob.Workers, *defaults)
+	podSpec, objectMeta, primaryContainerName, err := flytek8s.ToK8sPodSpec(ctx, taskCtx)
 	if err != nil {
 		return nil, err
 	}
-	schedulerSpec, err := createSchedulerSpec(*daskJob.Scheduler, clusterName, *defaults)
+	nonInterruptiblePodSpec := podSpec.DeepCopy()
+	removeInterruptibleConfig(nonInterruptiblePodSpec, taskCtx)
+
+	// Add labels and annotations to objectMeta as they're not added by ToK8sPodSpec
+	mergeMapInto(taskCtx.TaskExecutionMetadata().GetAnnotations(), objectMeta.Annotations)
+	mergeMapInto(taskCtx.TaskExecutionMetadata().GetLabels(), objectMeta.Labels)
+
```

Comment:
@hamersaw Is there a better way to do this? The `ObjectMeta` object which is returned does not have those set

Reply:
So on the top-levle object these are [set in flytepropeller](https://github.com/flyteorg/flytepropeller/blob/4322012ef07dff4de457473a7b5fb4cd6be4a2ee/pkg/controller/nodes/task/k8s/plugin_manager.go#L106-L107). Obviously this doesn't match the use-case here. The only alternative would be to take them out of flytepropeller, but then they would have to be added everywhere for each plugin and I'm sure it would break some users private plugins. I'm OK leaving as is if you are?

Reply:
I'm fine with leaving it 👍